### PR TITLE
Remove check for host IP from e2e test for replication controller

### PR DESCRIPTION
**Pull request by ghost**
This check has been replaced by the more targets #6705

### CODE REVIEW

**Comment by quinton-hoole** [comment]
Agreed.  I've changed the loops to retain the original PodList, relist, and fail if the new List operation returns something different.  I've also made the Logf and Failf messages more informative.  PTAL.

**Comment by quinton-hoole** [comment]
Regarding pod status, yes, that might be useful for debugging.  I'll just dump the status of all of the pods if the test fails.  Probably not worth the effort of doing anything more fancy than that.

**Comment by quinton-hoole** [comment]
OK, I added two separate commits, one of the refactor to use util.Poll, and the other for the rest. I don't particularly like the util.Poll solution as it's relatively messy to pass parameters into and out of the condition function, and in my mind makes the code less easy to understand.  But it's there now.  We can either merge it, or not.  I don't feel terribly strongly either way.  PTAL.

**Comment by quinton-hoole** [comment]
Ping.  It would be good to get this merged ASAP, as it's the only PR between us and a 100% clean e2e suite, and beautiful fields of green.

**Comment by bprashanth** [comment]
LGTM, also soo https://github.com/GoogleCloudPlatform/kubernetes/issues/6558#issuecomment-93066060 for hostip context

**Comment by quinton-hoole** [comment]
Thanks bprashanth@. All checks are green.  Can haz merge?

File: test/e2e/rc.go [hunk 1]
```diff
@@ -135,35 +137,43 @@ func ServeImageOrFail(c *client.Client, test string, image string) {
...
-			}
-			if time.Since(t) >= hostIPTimeout {
-				Failf("Controller %s: Gave up waiting for hostIP of replica %d after %v seconds",
-					name, i, time.Since(t).Seconds())
-			}
-			Logf("Controller %s: Retrying to get the hostIP of replica %d", name, i+1)
-			time.Sleep(5 * time.Second)
-		}
+	// Verify that something is listening.
+	By("Trying to dial each unique pod")
```
Comment (ixdy) [inline test/e2e/rc.go]: unimportant nit, but I noticed that you lowercased L131 but not this line. :)

Comment (quinton-hoole) [inline test/e2e/rc.go]: Done.  Thanks for spotting that.

**Comment by quinton-hoole** [comment]
Squashed and ready for merge.

**Comment by quinton-hoole** [comment]
Merging after verbal approval from bprashanth@

**Pull request merged**